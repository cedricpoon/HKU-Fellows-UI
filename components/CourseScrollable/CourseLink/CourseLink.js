import React, { Component } from 'react';
import { ListItem, Text, Left, Right, Icon, Content, View } from 'native-base';
import { Animated, Platform } from 'react-native';
import PropTypes from 'prop-types';
import NavigationService from 'hkufui/src/NavigationService';

import styles from '../Styles';

const fadeInDuration = 150;
const fadeInDelay = 50;

class CourseLink extends Component {

  constructor(props) {
    super(props);

    this.state = {
      fadeIn: new Animated.Value(0)
    }
  }

  componentDidMount() {
    Animated.timing(this.state.fadeIn, {
      toValue: 1,
      delay: fadeInDelay,
      duration : fadeInDuration
    }).start();
  }

  render() {
    const { title, description, onItemPress, bottomMost } = this.props;

    const context = (
      <Animated.View style={{ opacity: this.state.fadeIn }}>
        <ListItem
          style={[styles.linkItem, !bottomMost && styles.linkItemNonBottomMost]}
          onPress={() => {
            onItemPress();
            NavigationService.goBack();
          }}
        >
          <Left style={styles.linkLeft}>
            <Text numberOfLines={1} style={styles.headerText}>
              {title}
            </Text>
          </Left>
          <Content>
            <Text numberOfLines={1} style={[styles.linkDescription]}>
              {description}
            </Text>
          </Content>
          <Right style={styles.linkRight}>
            <Icon name={Platform.os === 'ios' ? "arrow-forward" : 'arrow-dropright'} style={[styles.headerText, styles.linkChevron]} />
          </Right>
        </ListItem>
      </Animated.View>
    );

    if (bottomMost)
      return (context);
    else
      return (
        <View style={styles.content}>
          {context}
        </View>
      );
  }
}

CourseLink.defaultProps = {
  onItemPress: () => {}
};

CourseLink.propTypes = {
  title: PropTypes.string.isRequired,
  description: PropTypes.string.isRequired,
  onItemPress: PropTypes.func,
  bottomMost: PropTypes.bool
};

export default CourseLink;
