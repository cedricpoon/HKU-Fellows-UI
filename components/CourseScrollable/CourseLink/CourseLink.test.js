import React from 'react';
import { shallow } from 'enzyme';

import CourseLink from './CourseLink';

describe('Testing CourseLink Component', () => {
  const defaultProps = {
    title: "CourseLink",
    description: "None"
  };

  it('renders as expected with required props', () => {
    const wrapper = shallow(
      <CourseLink {...defaultProps} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders as expected with required props and onItemPress', () => {
    const wrapper = shallow(
      <CourseLink {...defaultProps} onItemPress={() => {}} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders as expected in bottomMost with required props and onItemPress', () => {
    const wrapper = shallow(
      <CourseLink {...defaultProps} onItemPress={() => {}} bottomMost />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
