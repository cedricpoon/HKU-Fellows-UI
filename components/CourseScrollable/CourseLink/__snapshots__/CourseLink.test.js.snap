// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing CourseLink Component renders as expected in bottomMost with required props and onItemPress 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withNavigation(CourseLink)
    bottomMost={true}
    description="None"
    onItemPress={[Function]}
    title="CourseLink"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "Consumer": [Circular],
      "Provider": Object {
        "$$typeof": Symbol(react.provider),
        "_context": [Circular],
      },
      "_calculateChangedBits": null,
      "_currentRenderer": null,
      "_currentRenderer2": null,
      "_currentValue": undefined,
      "_currentValue2": undefined,
      "unstable_read": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": undefined,
        "_currentValue2": undefined,
        "unstable_read": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Testing CourseLink Component renders as expected with required props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withNavigation(CourseLink)
    description="None"
    title="CourseLink"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "Consumer": [Circular],
      "Provider": Object {
        "$$typeof": Symbol(react.provider),
        "_context": [Circular],
      },
      "_calculateChangedBits": null,
      "_currentRenderer": null,
      "_currentRenderer2": null,
      "_currentValue": undefined,
      "_currentValue2": undefined,
      "unstable_read": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": undefined,
        "_currentValue2": undefined,
        "unstable_read": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`Testing CourseLink Component renders as expected with required props and onItemPress 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <withNavigation(CourseLink)
    description="None"
    onItemPress={[Function]}
    title="CourseLink"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": [Function],
    },
    "ref": null,
    "rendered": [Function],
    "type": Object {
      "$$typeof": Symbol(react.context),
      "Consumer": [Circular],
      "Provider": Object {
        "$$typeof": Symbol(react.provider),
        "_context": [Circular],
      },
      "_calculateChangedBits": null,
      "_currentRenderer": null,
      "_currentRenderer2": null,
      "_currentValue": undefined,
      "_currentValue2": undefined,
      "unstable_read": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": [Function],
      },
      "ref": null,
      "rendered": [Function],
      "type": Object {
        "$$typeof": Symbol(react.context),
        "Consumer": [Circular],
        "Provider": Object {
          "$$typeof": Symbol(react.provider),
          "_context": [Circular],
        },
        "_calculateChangedBits": null,
        "_currentRenderer": null,
        "_currentRenderer2": null,
        "_currentValue": undefined,
        "_currentValue2": undefined,
        "unstable_read": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
